kuma/javascript/src/banners.jsx

```diff
@@ -211,6 +245,18 @@ export default function Banners(props: BannersProps) {
         let bannersList = props.banners || defaultBanners;
         for (const banner of bannersList) {
             if (userData.waffle.flags[banner.id]) {
+                // if this banner is to be shown to authenticated users only,
+                // and the current user is not authenticated, return `null`
+                if (banner.authenticated && !userData.isAuthenticated) {
+                    return null;
+                }
+
+                // special case for the subscriptions banner as we need access
+                // to the username as it forms part of the `url`
+                if (banner.id === 'mdn_subscription' && userData.username) {
```

Comment:
Partially in reference to https://github.com/mdn/kuma/issues/6577#issuecomment-595774644 I switched this back to using the `mdn_subscriptions` flag as there will be a period where we will show the form on edit profile, but not yet show the banner on other pages. This means we have a need for two independant switches.